### PR TITLE
[SVCS-579] Enhance CAS Login Pages for Institution Login

### PR DESCRIPTION
**Pull request by cslzchen**
### Ticket

https://openscience.atlassian.net/browse/SVCS-579

### Purpose

* Enhance institution login by making a button (similar to ORCiD) to replace the link.

### Side Effects

* Every customized pages in CAS needs an update.

### Changes

* Default Login Page

default-login

* Institution Login Page

inst-login

* Two-factor Auth Page

2fa-login

* OAuth Page

oauth-authorize

Please refer to the ticket for more pages (errors, internal, etc.)

### QA Notes

Test the following pages works as expected (links, buttons, language ...)

* Login Pages: default, institution, two-factor, oauth authorize page
* Login Failure Pages: account disabled, account not confirmed, remote login failed, oauth failed
* If you don't see the changes, you may need to delete the cache and do a hard reload

Note: I will test internal pages which is not available for users.

### Deployment Notes

No. Just deploy.

### CODE REVIEW

File: cas-server-support-osf/src/main/java/io/cos/cas/adaptors/postgres/handlers/OpenScienceFrameworkAuthenticationHandler.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ protected final HandlerResult authenticateInternal(final OpenScienceFrameworkCre
 
         // Check user's status, and only ACTIVE user can sign in
         if (USER_NOT_CONFIRMED.equals(userStatus)) {
-            throw new LoginNotAllowedException(username + " is not registered");
+            throw new LoginNotAllowedException(username + " is registered but not confirmed");
```
Comment (cslzchen) [inline cas-server-support-osf/src/main/java/io/cos/cas/adaptors/postgres/handlers/OpenScienceFrameworkAuthenticationHandler.java:184]: Fixed the wrong error description.

File: cas-server-webapp/src/main/resources/messages.properties [hunk 2]
```diff
@@ -26,16 +26,22 @@ screen.welcome.label.netid=<span class="accesskey">E</span>mail:
...
 screen.welcome.label.publicstation=I am at a public workstation.
 screen.welcome.label.warn=<span class="accesskey">W</span>arn me before logging me into other sites.
 screen.welcome.label.warn.accesskey=w
 screen.welcome.button.login=SIGN IN
+screen.welcome.button.login.institution=Sign in with Your Institution
 screen.welcome.button.login.orcid=Sign in with ORCID
 screen.welcome.button.clear=CLEAR
 screen.welcome.button.verify=VERIFY
 
+screen.general.link.createAccount=Create Account
```
Comment (cslzchen) [inline cas-server-webapp/src/main/resources/messages.properties]: Added or updated the messages for related pages. (Many occurrences.)

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casAccountDisabledView.jsp [hunk 3]
```diff
@@ -18,13 +18,28 @@
...
 
 <div id="msg" class="errors">
-    <h2><spring:message code="screen.accountdisabled.heading" /></h2>
-    <p><spring:message code="screen.accountdisabled.message" /></p>
+    <h2><spring:message code="screen.accountdisabled.heading"/></h2>
+    <p><spring:message code="screen.accountdisabled.message"/></p>
+
+    <hr/>
+    <br/>
+    <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casAccountDisabledView.jsp]: Added button for "Contact OSF Support" and removed the in-message link.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp [hunk 4]
```diff
@@ -34,29 +37,49 @@
...
-    <br>
+
+    <%-- OSF Username and Password Login --%>
+    <br/>
+    <hr/>
+    <br/>
+    <spring:eval var="osfLoginUrl" expression="@casProperties.getProperty('cas.osf.login.url')"/>
+    <c:set var="serviceParam"
+           value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
+    <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp]: Added a button for "Non-institution Login" and removed the link from the `bottom.jsp`.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginNotAllowedView.jsp [hunk 5]
```diff
@@ -18,14 +18,31 @@
...
 
-<jsp:directive.include file="includes/bottom.jsp" />
+<div id="msg" class="errors">
+
+    <h2><spring:message code="screen.loginnotallowed.heading"/></h2>
+    <p><spring:message code="screen.loginnotallowed.message"/></p>
+
+    <hr/>
+    <br/>
+    <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginNotAllowedView.jsp]: A good-looking button for "Resend Confirmation Email".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casOtpLoginView.jsp [hunk 6]
```diff
@@ -54,50 +57,39 @@
...
-            <input type="hidden" name="_eventId" value="submit" />
-            <input class="btn-submit" name="submit" accesskey="s" value="<spring:message code="screen.mfa.button.verify" />"  type="submit" />
-            <input class="btn-reset" name="cancel" accesskey="c"
-                   value="<spring:message code="screen.mfa.button.cancel" />" type="button"
-                   onclick="location.href = location.href;" />
-        </section> --%>
+        <%-- Cancel 2FA and Return to OSF --%>
+        <c:set var="serviceParam"
+               value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
+        <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casOtpLoginView.jsp]: Added a button for user to cancel and return to sign in page.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casOtpLoginView.jsp [hunk 7]
```diff
@@ -54,50 +57,39 @@
...
+            </a>
+        </section>
     </form:form>
 </div>
 
-<c:set var="alternativeBottomNone" value="true"/>
+<c:set var="linkSignIn" value="false"/>
+<c:set var="linkSignOut" value="false"/>
+<c:set var="linkCreateAccount" value="false"/>
+<c:set var="linkBackToOsf" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casOtpLoginView.jsp:90]: For 2FA page, only show "Back to OSF" at the bottom.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casRemoteUserFailedLoginView.jsp [hunk 8]
```diff
@@ -18,13 +18,19 @@
...
-<jsp:directive.include file="includes/bottom.jsp" />
+<div id="msg" class="errors">
+    <h2><spring:message code="screen.remoteuserfailedlogin.heading"/></h2>
+    <p><spring:message code="screen.remoteuserfailedlogin.message"/></p>
+</div>
+
+<c:set var="linkSignIn" value="false"/>
+<c:set var="linkSignOut" value="false"/>
+<c:set var="linkCreateAccount" value="false"/>
+<c:set var="linkBackToOsf" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casRemoteUserFailedLoginView.jsp:34]: For remote login failed page, only show "Back to OSF" at the bottom.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casShouldNotHappenView.jsp [hunk 9]
```diff
@@ -18,13 +18,29 @@
     under the License.
 
 --%>
-<jsp:directive.include file="includes/top.jsp" />
+
+<%--
+
+    OSF Login Exception Page -- Account with Inactive or Invalid Status:
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casShouldNotHappenView.jsp]: A fail-safe page which is impossible to reach through normal client flow.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp [hunk 10]
```diff
@@ -21,72 +21,49 @@
 
 <%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
 
-    <c:if test= "${empty alternativeBottomNone}">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp:24]: This logic is fully totally replaced.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp [hunk 11]
```diff
@@ -21,72 +21,49 @@
...
-            <br>
-            <spring:eval var="createAccountUrl" expression="@casProperties.getProperty('osf.createAccount.url')" />
-            <a id="create-account" href="${createAccountUrl}${registeredService.properties.registerUrl.getValue()}">Create Account</a>
-        </div>
-    </c:if>
+<c:if test="${empty serviceParam}">
+    <c:set var="serviceParam" value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
 </c:if>
 
+<div class="bottom-link">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp]: `bottom.js` now only takes care of the links under the grey login box: Sign In, Sign Out, Create Account, Back to OSF. Each page sets up the flags for what they want. No more complicated logic.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casShouldNotHappenView.jsp [hunk 12]
```diff
@@ -18,13 +18,29 @@
...
+    <p><spring:message code="screen.shouldnothappen.message"/></p>
 </div>
 
 <c:set var="alternativeBottomLogin" value="true"/>
 
-<jsp:directive.include file="includes/bottom.jsp" />
+<c:set var="linkSignIn" value="false"/>
+<c:set var="linkSignOut" value="false"/>
+<c:set var="linkCreateAccount" value="false"/>
+<c:set var="linkBackToOsf" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casShouldNotHappenView.jsp:44]: For this user inactive or invalid page, only show "Back to OSF" at the bottom.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLogoutView.jsp [hunk 13]
```diff
@@ -18,14 +18,21 @@
...
-    <p><spring:message code="screen.logout.success" /></p>
-    <p><spring:message code="screen.logout.security" /></p>
+    <h2><spring:message code="screen.logout.header"/></h2>
+    <br/>
+    <p><spring:message code="screen.logout.success"/></p>
+    <p><spring:message code="screen.logout.security"/></p>
 </div>
 
-<c:set var="alternativeBottomLogin" value="true"/>
+<c:set var="linkSignIn" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLogoutView.jsp:33]: For generic logged out page (when there is no service), show "Sign In", "Create Account" and "Back to OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casGenericSuccessView.jsp [hunk 14]
```diff
@@ -18,18 +18,25 @@
...
-    <p><spring:message code="screen.success.security" /></p>
-  </div>
-
-<c:set var="alternativeBottomLogout" value="true" />
+    <p><spring:message code="screen.success.security"/></p>
+</div>
 
-<jsp:directive.include file="includes/bottom.jsp" />
+<c:set var="linkSignIn" value="false"/>
+<c:set var="linkSignOut" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casGenericSuccessView.jsp:38]: For the generic logged in page, show "Sign Out" and "Back to OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casAccountDisabledView.jsp [hunk 15]
```diff
@@ -18,13 +18,28 @@
...
+        <a id="alt-login-inst" class="btn-alt-login" href="mailto:support@osf.io">
+            <img class="osf-alt-logo" src="../images/osf-alt-logo.png">
+            <span class="label-login"><spring:message code="screen.accountdisabled.button.contactSupport"/></span>
+        </a>
+    </section>
 </div>
 
-<c:set var="alternativeBottomLogin" value="true"/>
+<c:set var="linkSignIn" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casAccountDisabledView.jsp]: For the account disabled page, show "Sign In", "Create Account" and "Back to OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp [hunk 16]
```diff
@@ -59,4 +78,10 @@
         }
     </script>
 </div>
-<jsp:directive.include file="includes/bottom.jsp" />
+
+<c:set var="linkSignIn" value="false"/>
+<c:set var="linkSignOut" value="false"/>
+<c:set var="linkCreateAccount" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp:79]: For the institution login page, show "Create Account" and "Back to OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginNotAllowedView.jsp [hunk 17]
```diff
@@ -18,14 +18,31 @@
...
+        <spring:eval var="osfResendConfirmationUrl"
+                     expression="@casProperties.getProperty('osf.resendConfirmation.url')"/>
+        <a id="alt-login-inst" class="btn-alt-login" href="${osfResendConfirmationUrl}">
+            <img class="osf-alt-logo" src="../images/osf-alt-logo.png">
+            <span class="label-login"><spring:message code="screen.loginnotallowed.button.resendConfirmation"/></span>
+        </a>
+    </section>
+</div>
+
+<c:set var="linkSignIn" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginNotAllowedView.jsp:39]: For the account not confirmed page, show "Sign In", "Create Account" and "Back to OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp [hunk 18]
```diff
@@ -55,148 +57,83 @@
...
+        </section>
+
+        <%-- Institution Login --%>
+        <br/>
+        <hr/>
+        <br/>
+        <spring:eval var="institutionLoginUrl" expression="@casProperties.getProperty('cas.institution.login.url')"/>
+        <c:set var="serviceParam"
+               value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
+        <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp]: Added a button for institution login and removed the link from `bottom.jsp`.

**Review by cslzchen — COMMENTED** [review]
@icereval Ready for review.

I happened to use the "Format Code" in IntelliJ so half of the changes are style update. I pre-reviewed all the functional changes.

File: cas-server-webapp/src/main/webapp/css/cas.css [hunk 19]
```diff
@@ -29,7 +29,7 @@ abbr[title], dfn[title] { border-bottom: 1px dotted; cursor: help; }
 table { border-collapse: collapse; border-spacing: 0; }
 hr { display: block; height: 1px; border: 0; border-top: 1px solid #cccccc; margin: 0; padding: 0; }
 input, select { vertical-align: middle; }
-body { font-family: 'Open Sans'; font-size: 12px; line-height: 1.4em; background: #eee; }
+body { font-family: 'Open Sans'; font-size: 12px; line-height: 1.4em; background: #eee; min-width: 255px}
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/css/cas.css:32]: Added a minimal width for the body.

File: cas-server-webapp/src/main/webapp/css/cas.css [hunk 20]
```diff
@@ -69,25 +69,6 @@ h2 {
   text-align: center;
 }
 
-.title-abbr {
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/css/cas.css:72]: No longer needed due to switching from "Open Science Framework" to "OSF".

File: cas-server-webapp/src/main/webapp/css/cas.css [hunk 21]
```diff
@@ -201,8 +182,11 @@ header h1 {
 }
 
 #login .btn-oauth-submit {
+  display: inline-block;
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/css/cas.css:185]: Improved the appearance and effects of oauth "Allow" and "Deny" button.

File: cas-server-webapp/src/main/resources/messages.properties [hunk 22]
```diff
@@ -20,60 +20,67 @@
 #
 
 screen.welcome.welcome=Congratulations on bringing CAS online!  To learn how to authenticate, please review the default authentication handler configuration.
-screen.welcome.security=For security reasons, please Log Out and Exit your web browser when you are done accessing services that require authentication!
-screen.welcome.instructions=Enter your Username and Password
+screen.welcome.security=For security reasons, please log out and exit your web browser when you are done accessing services that require authentication!
```
Comment (cslzchen) [inline cas-server-webapp/src/main/resources/messages.properties:23]: In this file, updated all the messages, labels, etc., proposed by the Production and QA team.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casAccountDisabledView.jsp [hunk 23]
```diff
@@ -18,13 +18,19 @@
...
 
 <div id="msg" class="errors">
-    <h2><spring:message code="screen.accountdisabled.heading" /></h2>
-    <p><spring:message code="screen.accountdisabled.message" /></p>
+    <h2><spring:message code="screen.accountdisabled.heading"/></h2>
+    <p><spring:message code="screen.accountdisabled.message"/></p>
 </div>
 
-<c:set var="alternativeBottomLogin" value="true"/>
+<c:set var="linkSignIn" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casAccountDisabledView.jsp:31]: For the account disabled page, show "Sign In", "Create Account" and "Back to OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp [hunk 24]
```diff
@@ -43,12 +46,25 @@
...
     </section>
-    <br>
+
+    <%-- OSF Username and Password Login --%>
+    <br/>
+    <hr/>
+    <br/>
+    <spring:eval var="osfLoginUrl" expression="@casProperties.getProperty('cas.osf.login.url')"/>
+    <c:set var="serviceParam" value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
+    <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp]: Added a button for OSF login.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginNotAllowedView.jsp [hunk 25]
```diff
@@ -18,14 +18,27 @@
...
 
-<c:set var="alternativeBottomLogin" value="true"/>
+<jsp:directive.include file="includes/top.jsp"/>
 
-<jsp:directive.include file="includes/bottom.jsp" />
+<div id="msg" class="errors">
+    <h2><spring:message code="screen.loginnotallowed.heading"/></h2>
+    <p><spring:message code="screen.loginnotallowed.message"/></p>
+    <hr/><br/>
+    <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginNotAllowedView.jsp:30]: Added a button for resend confirmation email.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp [hunk 26]
```diff
@@ -19,13 +19,15 @@
 
 --%>
 
-<jsp:directive.include file="includes/top.jsp" />
+<%-- OSF CAS default login page --%>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp:22]: In this file, removed unused / irrelevant code which has been commented out.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp [hunk 27]
```diff
@@ -55,148 +57,73 @@
...
-            <a id="forgot-password" class='need-help' href="${forgotPasswordUrl}" title="<spring:message code="logo.title" />">Forgot Your Password?</a>
+            <input type="checkbox" name="rememberMe" id="rememberMe" value="true" checked tabindex="5"/>
+            <label for="rememberMe"><spring:message code="screen.rememberme.checkbox.title"/></label>
+            <spring:eval var="forgotPasswordUrl" expression="@casProperties.getProperty('osf.forgotPassword.url')"/>
+            <a id="forgot-password" class='need-help' href="${forgotPasswordUrl}" title="<spring:message code="logo.title" />"><spring:message code="screen.general.link.forgotPassword"/></a>
+        </section>
+
+        <%-- Institution Login --%>
+        <br/><hr/><br/>
+        <spring:eval var="institutionLoginUrl" expression="@casProperties.getProperty('cas.institution.login.url')"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp:96]: Added a button for institution login (similar to ORCiD login).

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp [hunk 28]
```diff
@@ -55,148 +57,73 @@
...
-                        <li class="last"><a href="${loginUrl}pl">Polish</a></li>
-                    </ul>
-                </c:otherwise>
-            </c:choose>
-        </div>
-    </div>
-</div> --%>
+<c:set var="linkSignIn" value="false"/>
+<c:set var="linkSignOut" value="false"/>
+<c:set var="linkCreateAccount" value="true"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp:126]: For the default login page, show "Create Account" and "Back to OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casOtpLoginView.jsp [hunk 29]
```diff
@@ -54,50 +57,36 @@
...
-            <input type="hidden" name="execution" value="${flowExecutionKey}" />
-            <input type="hidden" name="_eventId" value="submit" />
-            <input class="btn-submit" name="submit" accesskey="s" value="<spring:message code="screen.mfa.button.verify" />"  type="submit" />
-            <input class="btn-reset" name="cancel" accesskey="c"
-                   value="<spring:message code="screen.mfa.button.cancel" />" type="button"
-                   onclick="location.href = location.href;" />
-        </section> --%>
+        <%-- Cancel 2FA and Return to OSF --%>
+        <c:set var="serviceParam" value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
+        <section class="row">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casOtpLoginView.jsp:77]: Added a button for user to cancel and return to the sign in page.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp [hunk 30]
```diff
@@ -21,112 +21,88 @@
...
-<c:if test= "${empty alternativeBottomNone}">
-    <c:if test="${empty alternativeBottomLogout}">
-        <div class="row" style="text-align: center;">
-            <br>
-            <spring:eval var="createAccountUrl" expression="@casProperties.getProperty('osf.createAccount.url')" />
-            <a id="create-account" href="${createAccountUrl}${registeredService.properties.registerUrl.getValue()}">Create Account</a>
-        </div>
+<div class="bottom-link">
+    <br/>
+    <c:if test="${linkSignIn}">
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp:33]: `bottom.jsp` only handles four redirect links and the copyright footer now. No more complicated logic.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/top.jsp [hunk 31]
```diff
@@ -68,13 +68,8 @@
...
                                 <span>OSF Institutions</span>
                             </c:when>
-                            <c:when test="${not empty registeredService}">
-                                <span class="title-full">${registeredService.properties.title.getValue()}</span>
-                                <span class="title-abbr">${registeredService.properties.titleAbbr.getValue()}</span>
-                            </c:when>
                             <c:otherwise>
-                                <span class="title-full">Open Science Framework</span>
-                                <span class="title-abbr">OSF CAS</span>
+                                <span>OSF</span>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/top.jsp:72]: Now every page shows "OSF".

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/protocol/oauth/oAuthFailureView.jsp [hunk 32]
```diff
@@ -18,14 +18,17 @@
...
-    <h2><spring:message code="screen.oauth.error.header" /></h2>
-    <p><spring:message code="${rootCauseException.code}" /></p>
-    <p><spring:message code="${rootCauseException.message}" /></p>
+    <h2><spring:message code="screen.oauth.error.header"/></h2>
+    <p><spring:message code="screen.oauth.error.message"/></p>
+    <p><spring:message code="${rootCauseException.code}"/><br/><spring:message code="${rootCauseException.message}"/></p>
 </div>
 
-<c:set var="alternativeBottomLogout" value="true"/>
+<c:set var="linkSignIn" value="false"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/protocol/oauth/oAuthFailureView.jsp:29]: Show "Sign out" and "Back to OSF" on OAuth failure page.

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/protocol/oauth/confirm.jsp [hunk 33]
```diff
@@ -18,31 +18,40 @@
...
+            <input type="submit" class="btn-oauth-submit" name="action" accesskey="a"
+                   value="<spring:message code="screen.oauth.confirm.allow" />"/>
+            <input type="submit" class="btn-oauth-reset" name="action" accesskey="d"
+                   value="<spring:message code="screen.oauth.confirm.deny" />"/>
         </section>
     </form>
 </div>
 
-<c:set var="alternativeBottomLogout" value="true"/>
+<c:set var="linkSignIn" value="false"/>
```
Comment (cslzchen) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/protocol/oauth/confirm.jsp:52]: Don't show any links for the OAuth authorize page.

**Review by cslzchen — COMMENTED** [review]
Responded to product demo and ready for CR again. 🎆 🎆

**Review by icereval — APPROVED** [review]
LGTM

**Pull request merged**